### DISCUSSION TITLE
Inquiry about Capturing Network Traffic on DPDK Interface

### DISCUSSION THREAD

**Discussion by shashank-6777** [netobserv/network-observability-operator#405] [General]
Dear Team,

I hope this message finds you well. I'm currently exploring the possibility of capturing network traffic on a DPDK interface within our OpenShift environment. I believe this could provide valuable insights into our network performance and troubleshooting efforts as most of the Telco application (user plane) used DPDK interfaces.

**Is it possible to capture network traffic on a DPDK interface in our OpenShift setup using this operator ?**

Thank you for your time and expertise. Looking forward to your input.

**Comment by jotak** [comment]
Hello @shashank-6777 ,
I must admit I am not familiar at all with DPDK but I asked some engineers around , if I understand correctly with dpdk the kernel network stack is bypassed, which would not allow netobserv to capture flows since it is hooking at the tc level. So this wouldn't work.
Can I ask you, is it a specific CNI that you use and that uses dpdk? I'm not sure if CNIs such as openshift-sdn or ovn can be configured to use dpdk, I don't think so (?)

**Comment by shashank-6777** [comment]
Hi @jotak,

Thanks for your response,

Your understanding is correct, DPDK bypasses the kernel network stack, tools that rely on kernel-level hooks, like netobserv which hooks into the tc (traffic control) layer, may indeed face challenges in capturing and monitoring network flows when DPDK is in use.

We have installed our cluster by using OVN as CNI and we are injecting secondary interfaces (VF"s) using multus as vfio-pci device type (Intel xxv710 nic card).

I understand from your explanation that capturing network traffic in a DPDK-based environment can be challenging due to the bypassing of the kernel network stack, but the thing is customer wants to monitor the traffic for all interfaces in the application pod i.e OVN/IPVLAN/MacVlan (able to capture) but the DPDK interface plays a crucial role in achieving high-performance data processing in 5G networks. Monitoring this interface allows you to identify performance bottlenecks, latency issues, and resource utilization patterns.

Thank you for your time and expertise. Looking forward to your input.

**Reply by jotak** [reply]
any idea if the dpdk plugin is able to export flows in NetFlows or IPFIX format? Netobserv is able to collect IPFIX flows, although it has been deprecated because our eBPF agent provides better performances, essentially thanks to avoiding too many kernel-userspace copies ... but the context with DPDK is different and it seems like Netflows would make more sense here. It might be a good reason to bring back our netflows collection full support, if it can help

**Comment by shashank-6777** [comment]
Hi @jotak,

Thanks for your response,

Sorry but as per my knowledge DPDK itself does not natively provide plugins or libraries specifically designed to export flows in NetFlow or IPFIX format. DPDK is primarily focused on providing libraries and tools for high-performance packet processing in user space, and flow export is often not within its core scope.

i think since DPDK interfaces operate in user space, you might need to tap the traffic flowing through those interfaces. One approach could be using technologies like eBPF (already in use by your operator) to tap into the network stack and capture relevant packets, Cilium, which is often used with DPDK and Kubernetes, provides eBPF-based networking and security features that could be useful here. sorry but am not a expert may be am wrong.

`It might be a good reason to bring back our netflows collection full support, if it can help`
It really helps to all the cloud platform vendors because as per my search there is no opensource tool by which we can monitor DPDK interfaces in kubernetes/openshift. Also help people like us to push customer to use this tool for there edge infra structure.

Please let me know you expert advice and suggestion.

Thanks in advance

**Reply by jotak** [reply]
Hi @shashank-6777 
There might not be any short term answer for supporting DPDK with netobserv but I definitely get your point and subscribe. We're starting some discussions around that internally. Thanks for bringing up this topic. I'll let you know when I have some updates.

**Reply by shashank-6777** [reply]
Thank you for your prompt response @jotak 
It's great to hear that you're initiating discussions internally to address this topic. I look forward to hearing more about it when you have some updates. If there's any additional information or input I can provide to assist in these discussions, please feel free to reach out.

Once again, thank you for your attention to this matter, and I'm eager to see how this evolves.

**Reply by shashank-6777** [reply]
Hi @jotak,

I hope all is well. Just wanted to check if you have any update.  
Your assistance is greatly appreciated. Thank you for your time and attention to this matter.

**Reply by jotak** [reply]
hey @shashank-6777 , fyi, we plan starting a PoC soon: https://issues.redhat.com/browse/NETOBSERV-1272

**Reply by shashank-6777** [reply]
@jotak, That's great to hear! 
Your assistance is greatly appreciated.

**Reply by shashank-6777** [reply]
Hi @jotak is there any update on this ?

**Reply by shashank-6777** [reply]
hi @jotak , How are your doing.

Is this RFP is open or drop ?
If possible could you please update if you and your team are working.

**Reply by jotak** [reply]
Hey @shashank-6777 , sorry for taking that long to answer. There are technical challenges we are facing and we don't have clear plans of how to address that. That doesn't mean this is dropped, I hope we'll be able to address that at some point, but I'm afraid that's not something coming anytime soon.
